sdk/identity/azure-identity/src/token_credential_impl.cpp

```diff
@@ -87,103 +108,193 @@ Azure::Core::Credentials::AccessToken TokenCredentialImpl::GetToken(
         request = shouldRetry(statusCode, *response);
         if (request == nullptr)
         {
-          std::ostringstream errorMsg;
-          errorMsg << errorMsgPrefix << "error response: "
-                   << static_cast<std::underlying_type<HttpStatusCode>::type>(statusCode) << " "
-                   << response->GetReasonPhrase();
-
-          throw AuthenticationException(errorMsg.str());
+          throw std::runtime_error(
```

Comment:
Why not throw a RequestFailedException here?

Reply:
We wrap it later. And it should not be RequestFailedException, but must be AuthenticationException. Because otherwise, when thrown from the client pipeline, you won't be able to distinguish between service errors and authentication errors.